modules/reindex/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecReindexIT.java

```diff
@@ -0,0 +1,189 @@
+                .collect(Collectors.toSet())
+                .containsAll(codecMap.values())
+        );
+
+        // creating destination index with destination codec
+        createIndex(
+            destIndex,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put("index.codec", destCodec)
+                .build()
+        );
+
+        BulkByScrollResponse bulkResponse = new ReindexRequestBuilder(client(), ReindexAction.INSTANCE).source(index)
+            .destination(destIndex)
+            .refresh(true)
+            .waitForActiveShards(ActiveShardCount.ONE)
+            .get();
+
+        assertEquals(codecMap.size() * nbDocs, bulkResponse.getCreated());
+        assertEquals(codecMap.size() * nbDocs, bulkResponse.getTotal());
+        assertEquals(0, bulkResponse.getDeleted());
+        assertEquals(0, bulkResponse.getNoops());
+        assertEquals(0, bulkResponse.getVersionConflicts());
+        assertEquals(1, bulkResponse.getBatches());
+        assertTrue(bulkResponse.getTook().getMillis() > 0);
+        assertEquals(0, bulkResponse.getBulkFailures().size());
+        assertEquals(0, bulkResponse.getSearchFailures().size());
+        assertTrue(getSegments(destIndex).stream().allMatch(segment -> segment.attributes.containsValue(destCodecMode)));
+    }
+
+    private void useCodec(String index, String codec) throws ExecutionException, InterruptedException {
+        assertAcked(client().admin().indices().prepareClose(index));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index));
+    }
+
+    private void flushAndRefreshIndex(String index) {
+
+        // Request is not blocked
+        for (String blockSetting : Arrays.asList(
+            SETTING_BLOCKS_READ,
+            SETTING_BLOCKS_WRITE,
+            SETTING_READ_ONLY,
+            SETTING_BLOCKS_METADATA,
+            SETTING_READ_ONLY_ALLOW_DELETE
+        )) {
+            try {
+                enableIndexBlock(index, blockSetting);
+                // flush
+                FlushResponse flushResponse = client().admin().indices().prepareFlush(index).setForce(true).execute().actionGet();
+                assertNoFailures(flushResponse);
+
+                // refresh
+                RefreshResponse refreshResponse = client().admin().indices().prepareRefresh(index).execute().actionGet();
+                assertNoFailures(refreshResponse);
+            } finally {
+                disableIndexBlock(index, blockSetting);
+            }
+        }
+    }
+
+    private void ingestUsingCodec(String index, int nbDocs) throws InterruptedException {
```

Comment:
nit: There is no codec involved here, it is just ingest. Can be renamed to ```insertDocs```

Reply:
done